### PR TITLE
File persistance

### PR DESCRIPTION
**Pull request by brtkbaranowsky**
@piotr-grundszok You forgot to create a pull request. This time I will do it for you.

### CODE REVIEW

File: src/com/grundszok/piotr/app/App.java [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
 
 class App {
     public static void main(String[] args) {
+        PersistanceService persistanceService = new PersistanceService();
         DisplayService displayService = new AstrixDisplayService();
         InputService inputService = new InputService(displayService);
-        MenuController menuController = new MenuController(displayService, inputService);
-
-        System.out.println("Git test");
+        MenuController menuController = new MenuController(displayService, inputService, persistanceService);
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/App.java]: Can be initialized directly in Menu Controller constructor

File: src/com/grundszok/piotr/app/controllers/MenuController.java [hunk 2]
```diff
@@ -14,12 +16,14 @@
 
     DisplayService displayService;
     InputService inputService;
+    PersistanceService persistanceService;
 
     List<Item> shoppingList;
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/controllers/MenuController.java]: all of those can be private

File: src/com/grundszok/piotr/app/controllers/MenuController.java [hunk 3]
```diff
@@ -33,7 +37,7 @@ public void chooseDisplayStrategyFromUserInput() {
                 "Dostepne opcje: + , = , *\n" +
                 "Zeby wyjsc wproadz: X";
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/controllers/MenuController.java]: Move string literals to separate class

File: src/com/grundszok/piotr/app/controllers/MenuController.java [hunk 4]
```diff
@@ -33,7 +37,7 @@ public void chooseDisplayStrategyFromUserInput() {
...
 
-        print(msg);
+        displayService.print(msg);
         char displayCharFromUser = inputService.getDisplayCharFromUser();
         while (displayCharFromUser != 'X') {
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/controllers/MenuController.java]: magic string - move to variable and give it proper name

File: src/com/grundszok/piotr/app/controllers/MenuController.java [hunk 5]
```diff
@@ -43,7 +47,7 @@ public void chooseDisplayStrategyFromUserInput() {
                 displayService.print("Unknown character for display type");
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/controllers/MenuController.java]: same as above

File: src/com/grundszok/piotr/app/controllers/MenuController.java [hunk 6]
```diff
@@ -58,42 +62,51 @@ public void start() {
...
                 case "4" -> chooseDisplayStrategyFromUserInput();
+                case "5" -> {
+                    try {
+                        persistanceService.save(shoppingList);
+                        displayService.print(String.format("Zapisano %d elementów do pliku", shoppingList.size()));
+                    } catch (EmptyShoppingListException exception) {
+                        displayService.print(exception.getMessage());
+                    }
+                }
                 case "X" -> System.exit(0);
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/controllers/MenuController.java]: Move strings to proper class and refactor cases to use private methods for readability sake

File: src/com/grundszok/piotr/app/controllers/MenuController.java [hunk 7]
```diff
@@ -58,42 +62,51 @@ public void start() {
...
 
     private void printMenu() {
-        print("1. Stworz liste zakupow\n"
+        displayService.print("1. Stworz liste zakupow\n"
                 + "2. Wyświetl liste zakupow\n"
                 + "3. Wyczyść liste zakupow\n"
-                + "4. Zmien styl wyświetlania\n\n"
+                + "4. Zmien styl wyświetlania\n"
+                + "5. Zapisz listę zakupow do pliku\n\n"
                 + "X. Zakoncz program");
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/controllers/MenuController.java]: Same as above

File: src/com/grundszok/piotr/app/services/PersistanceService.java [hunk 8]
```diff
@@ -1,4 +1,28 @@
...
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
 public class PersistanceService {
+    public void save(List<Item> shoppingList) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("HH-mm-ss");
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/services/PersistanceService.java]: could be class variable

File: src/com/grundszok/piotr/app/services/PersistanceService.java [hunk 9]
```diff
@@ -1,4 +1,28 @@
...
+        try (FileWriter fileWriter = new FileWriter("ShoppingList_" + LocalDateTime.now().format(dateTimeFormatter) + ".txt")) {
+            for (Item item : shoppingList
+            ) {
+                fileWriter.write(item.toString());
+                fileWriter.write('\n');
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/services/PersistanceService.java]: could be one lined

File: src/com/grundszok/piotr/app/services/PersistanceService.java [hunk 10]
```diff
@@ -1,4 +1,28 @@
...
+            }
+        } catch (IOException exception) {
+            exception.printStackTrace();
```
Comment (brtkbaranowsky) [inline src/com/grundszok/piotr/app/services/PersistanceService.java]: handle exception properly! user doesn't need to know about stacktrace

File: src/com/grundszok/piotr/app/App.java (hunk 1 above)
Comment (piotr-grundszok) [reply src/com/grundszok/piotr/app/App.java]: Ok, but i will leave DisplayServivce as variable

**Pull request merged**